build.sh

```diff
@@ -42,94 +38,20 @@ if [ -z "$DESTDIR" ]; then
   DESTDIR=$(mktemp -d)
 fi
 
-function build_otel() {
-  cd submodules/opentelemetry-java-contrib
-  mkdir -p "$DESTDIR$subagentdir/opentelemetry-collector/"
-  ./gradlew --no-daemon :jmx-metrics:build
-  cp jmx-metrics/build/libs/opentelemetry-jmx-metrics-*-SNAPSHOT.jar "$DESTDIR$subagentdir/opentelemetry-collector/opentelemetry-java-contrib-jmx-metrics.jar"
-
-  # Rename LICENSE file because it causes issues with file hash consistency due to an unknown
-  # issue with the debuild/rpmbuild processes. Something is unzipping the jar in a case-insensitive
-  # environment and having a conflict between the LICENSE file and license/ directory, leading to a changed jar file
-  mkdir ./META-INF
-  unzip -j "$DESTDIR$subagentdir/opentelemetry-collector/opentelemetry-java-contrib-jmx-metrics.jar" "META-INF/LICENSE" -d ./META-INF
-  zip -d "$DESTDIR$subagentdir/opentelemetry-collector/opentelemetry-java-contrib-jmx-metrics.jar" "META-INF/LICENSE"
-  mv ./META-INF/LICENSE ./META-INF/LICENSE.renamed
-  zip -u "$DESTDIR$subagentdir/opentelemetry-collector/opentelemetry-java-contrib-jmx-metrics.jar" "META-INF/LICENSE.renamed"
-
-  cd ../opentelemetry-operations-collector
-  # Using array assignment to drop the filename from the sha256sum output
-  JAR_SHA_256=($(sha256sum "$DESTDIR$subagentdir/opentelemetry-collector/opentelemetry-java-contrib-jmx-metrics.jar"))
-  go build -buildvcs=false -o "$DESTDIR$subagentdir/opentelemetry-collector/otelopscol" \
-    -ldflags "-X github.com/open-telemetry/opentelemetry-collector-contrib/receiver/jmxreceiver.MetricsGathererHash=$JAR_SHA_256" \
-    ./cmd/otelopscol
-}
-
-function build_fluentbit() {
-  cd submodules/fluent-bit
-  mkdir -p build
-  cd build
-  # CMAKE_INSTALL_PREFIX here will cause the binary to be put at
-  # /usr/lib/google-cloud-ops-agent/bin/fluent-bit
-  # Additionally, -DFLB_SHARED_LIB=OFF skips building libfluent-bit.so
-  cmake .. -DCMAKE_INSTALL_PREFIX=$subagentdir/fluent-bit \
-    -DFLB_HTTP_SERVER=ON -DFLB_DEBUG=OFF -DCMAKE_BUILD_TYPE=RelWithDebInfo \
-    -DWITHOUT_HEADERS=ON -DFLB_SHARED_LIB=OFF -DFLB_STREAM_PROCESSOR=OFF \
-    -DFLB_MSGPACK_TO_JSON_INIT_BUFFER_SIZE=1.5 -DFLB_MSGPACK_TO_JSON_REALLOC_BUFFER_SIZE=.10 \
-    -DFLB_CONFIG_YAML=OFF
-  make -j8
-  make DESTDIR="$DESTDIR" install
-  # We don't want fluent-bit's service or configuration, but there are no cmake
-  # flags to disable them. Prune after build.
-  rm "${DESTDIR}/lib/systemd/system/fluent-bit.service"
-  rm -r "${DESTDIR}${subagentdir}/fluent-bit/etc"
-}
-
 function build_opsagentengine() {
-  mkdir -p "$DESTDIR$prefix/libexec"
-  go build -buildvcs=false -o "$DESTDIR$prefix/libexec/google_cloud_ops_agent_engine" \
+  if [[ ! -f /work/google_cloud_ops_agent_engine ]]; then
+  go build -buildvcs=false -o "/work/google_cloud_ops_agent_engine" \
```

Comment:
@hsmatulisgoogle This changes break the build.sh if its run directly. For example,

```
$ BUILD_DISTRO=arch CODE_VERSION="${pkgver}" DESTDIR="$_destdir"  ./build.sh
```

See build output here: https://build.kilabit.info/karajo/job/log?id=aur_google-cloud-ops-agent-git&counter=51

Reply:
> build.sh is an implementation detail of Dockerfile and is not intended to be run standalone.

I see. I thought it was for building the binaries (the usual `make` flow installed on `$DESTDIR`). I remember reading the rpm spec script and it execute the `build.sh` directly.

> Would you mind sharing the reason why you're building it that way instead of using Dockerfile -- is there something about Dockerfile that doesn't work for you?

I use it to build the ops-agent package for Arch Linux [1] which is not available in Dockerfile.
In that case I will patch the build script manually, and run each .sh script manually.

```
@@ -72,6 +74,13 @@ build() {
        echo "destdir: $_destdir"

        cd "${pkgname}"
+
+       CGO_ENABLED=1 ./builds/otel.sh "$_destdir"
+       ./builds/fluent_bit.sh "$_destdir"
+       ./builds/systemd.sh "$_destdir"
+       ./builds/ops_agent_diagnostics.sh "$_destdir"
+
+       git apply "${srcdir}/0001-build-sh.patch"
        BUILD_DISTRO=arch CODE_VERSION="${pkgver}" DESTDIR="$_destdir" \
                ./build.sh
 }
```
It succesfully build last time I run, but need to test it on live image.

Sorry for the noise.

[1] https://aur.archlinux.org/packages/google-cloud-ops-agent-git